### PR TITLE
Install-script changes

### PR DESCRIPTION
**Pull request by ghost**
Miscellaneous script fixes...

Addresses:
https://github.com/dotnet/cli/issues/6995
    scripts/obtain/dotnet-install.ps1 - lines 483 & 492
https://github.com/dotnet/cli/issues/7367
    scripts/obtain/dotnet-install.sh  - line 605

Other minor changes...

### CODE REVIEW

File: scripts/obtain/dotnet-install.sh [hunk 1]
```diff
@@ -583,7 +584,7 @@ downloadcurl() {
 
     local failed=false
     if [ -z "$out_path" ]; then
-        curl --retry 10 -sSL -f --create-dirs $remote_path || failed=true
+        curl --retry 10 -sSL -f --create-dirs -o $remote_path || failed=true
```
Comment (wfurt) [inline scripts/obtain/dotnet-install.sh]: I don't think this is right. -o $remore_path will try to output of $remote_path. 
As far as I could tell, curl path was always right as curl would output to stdout by default.

File: scripts/obtain/dotnet-install.sh [hunk 2]
```diff
@@ -601,7 +602,7 @@ downloadwget() {
 
     local failed=false
     if [ -z "$out_path" ]; then
-        wget -q --tries 10 $remote_path || failed=true
+        wget -q --tries 10 -O $remote_path || failed=true
```
Comment (wfurt) [inline scripts/obtain/dotnet-install.sh]: I think the change should be '-O - ' 
('-' stands for stdout)

File: scripts/obtain/dotnet-install.ps1 [hunk 3]
```diff
@@ -449,14 +449,14 @@ function Prepend-Sdk-InstallRoot-To-Path([string]$InstallRoot, [string]$BinFolde
 
 $CLIArchitecture = Get-CLIArchitecture-From-Architecture $Architecture
 $SpecificVersion = Get-Specific-Version-From-Version -AzureFeed $AzureFeed -Channel $Channel -Version $Version
-$DownloadLink = Get-Download-Link -AzureFeed $AzureFeed -Channel $Channel -SpecificVersion $SpecificVersion -CLIArchitecture $CLIArchitecture
```
Comment (wli3) [inline scripts/obtain/dotnet-install.ps1]: So there is no channel anymore?

Comment (johnbeisner) [reply scripts/obtain/dotnet-install.ps1]: Not for this call - the channel has been used to determine the version, therefore is no longer needed at this point.

**Review by wli3 — APPROVED** [review]

**Pull request merged**